### PR TITLE
TPWDCEE-6243

### PR DESCRIPTION
**Pull request by sebastijansakac**

### CODE REVIEW

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/helper/CurrencyHelper.php  2
+ wirecardpaymentgateway/controllers/admin/WirecardTransactions.php  2
+ wirecardpaymentgateway/helper/TransactionBuilder.php  1
+ wirecardpaymentgateway/classes/Response/Failure.php  1
+ wirecardpaymentgateway/classes/Response/Success.php  2
+ wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  3
+ wirecardpaymentgateway/classes/Service/OrderAmountCalculatorService.php  1
+ wirecardpaymentgateway/classes/Notification/Failure.php  2
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/helper/OrderStateTransferObject.php  -1
- wirecardpaymentgateway/models/Transaction.php  -1
- wirecardpaymentgateway/helper/OrderManager.php  -1
- wirecardpaymentgateway/classes/Service/OrderStateManagerService.php  -9
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5633498&bid=18449736)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/controllers/admin/WirecardTransactions.php  2
+ wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  3
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/models/Transaction.php  -1
- wirecardpaymentgateway/helper/OrderStateTransferObject.php  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5633498&bid=18449736)

File: wirecardpaymentgateway/helper/NumericHelper.php [hunk 1]
```diff
@@ -5,6 +5,35 @@
...
+    public function getPsPrecision()
+    {
+        return (int)\Configuration::get('PS_PRICE_DISPLAY_PRECISION');
+    }
+
+    /**
+     * Return step for postprocessing amount input field
+     * @return string
+     */
+    public function getStep()
```
Comment (edra-bumci) [inline wirecardpaymentgateway/helper/NumericHelper.php:21]: @sebastijansakac is getStep() a proper name for what this function is doing?

File: wirecardpaymentgateway/helper/NumericHelper.php [hunk 2]
```diff
@@ -5,6 +5,35 @@
...
+    {
+        $precision = $this->getPsPrecision();
+        $step = '1';
+        if ($precision > 0) {
+            $step = '';
+            for ($i = 0; $i < $precision; $i++) {
```
[the remark this answers is not in the document]
Comment (sebastijansakac) [reply wirecardpaymentgateway/helper/NumericHelper.php]: @edra-bumci Good catch, fixed

File: wirecardpaymentgateway/helper/NumericHelper.php (hunk 1 above)
Comment (sebastijansakac) [reply wirecardpaymentgateway/helper/NumericHelper.php]: renamed to calculateNumericInputStep()

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/classes/Service/OrderAmountCalculatorService.php  1
+ wirecardpaymentgateway/classes/Response/Failure.php  1
+ wirecardpaymentgateway/controllers/admin/WirecardTransactions.php  2
+ wirecardpaymentgateway/helper/CurrencyHelper.php  2
+ wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  3
+ wirecardpaymentgateway/classes/Response/Success.php  2
+ wirecardpaymentgateway/helper/TransactionBuilder.php  1
+ wirecardpaymentgateway/classes/Notification/Failure.php  2
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/helper/NumericHelper.php  -3
- wirecardpaymentgateway/helper/OrderStateTransferObject.php  -1
- wirecardpaymentgateway/classes/Service/OrderStateManagerService.php  -9
- wirecardpaymentgateway/helper/OrderManager.php  -1
- wirecardpaymentgateway/models/Transaction.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5633498&bid=18449736)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/controllers/admin/WirecardTransactions.php  2
+ wirecardpaymentgateway/helper/TransactionBuilder.php  1
+ wirecardpaymentgateway/helper/CurrencyHelper.php  2
+ wirecardpaymentgateway/classes/Service/OrderAmountCalculatorService.php  1
+ wirecardpaymentgateway/classes/Response/Success.php  2
+ wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  3
+ wirecardpaymentgateway/classes/Notification/Failure.php  2
+ wirecardpaymentgateway/classes/Response/Failure.php  1
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/helper/NumericHelper.php  -3
- wirecardpaymentgateway/helper/OrderManager.php  -1
- wirecardpaymentgateway/classes/Service/OrderStateManagerService.php  -9
- wirecardpaymentgateway/models/Transaction.php  -1
- wirecardpaymentgateway/helper/OrderStateTransferObject.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5633498&bid=18449736)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  3
+ wirecardpaymentgateway/controllers/admin/WirecardTransactions.php  2
         

Coverage decreased per file
===========================
- wirecardpaymentgateway/helper/OrderStateTransferObject.php  -2
- wirecardpaymentgateway/models/Transaction.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5633498&bid=18449736)

**Comment by rinnhofer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ wirecardpaymentgateway/controllers/admin/WirecardTransactions.php  2
+ wirecardpaymentgateway/classes/Transaction/Builder/PostProcessingTransactionBuilder.php  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/prestashop-ee/pullRequest?prid=5633498&bid=18449736)

**Pull request merged**